pkg/services/ngalert/schedule/schedule.go

```diff
@@ -380,15 +380,16 @@ func (sch *schedule) ruleRoutine(grafanaCtx context.Context, key ngmodels.AlertR
 		ruleEval, err := sch.evaluatorFactory.Create(evalCtx, e.rule.GetEvalCondition())
 		var results eval.Results
 		var dur time.Duration
-		if err == nil {
+		if err != nil {
```

Comment:
The inverted logic better highlights the issue visually, IMO. Both paths are a little more symmetrical, you can see `dur` calculated in both, etc